include/boost/url/rfc/detail/impl/hier_part_rule.ipp

```diff
@@ -65,7 +65,13 @@ parse(
         t.authority = *rv;
         t.has_authority = true;
     }
-    if(it == end)
+    // the authority requires an absolute path
+    // or an empty path
+    if(it == end ||
+       (t.has_authority &&
+         (*it != '/' &&
+          *it != '?' &&
+          *it != '#')))
```

Comment:
This fix needs to go into develop as its own commit?

Reply:
This minor fix was getting in the way of this PR so I fixed it. I can not squash this commit if you want. Or undo it at the end open another PR.

Reply:
Its fine to fix it in the PR and also submit a separate PR to patch it in develop, git is smart enough to elide it when you squash and rebase.